src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java

```diff
@@ -260,4 +264,124 @@ public void validateRecordsTransferredToTargetTableIsEqualToNumberOfRecordsFromS
+              + "." + TestSetupHooks.bqSourceTable;
+    } else if (property.equalsIgnoreCase("partitionEndDate")) {
+      tableFullName = PluginPropertyUtils.pluginProp(value) + ":"
+              + PluginPropertyUtils.pluginProp("partitionTo")
+              + "." + TestSetupHooks.bqSourceTable;
+    }
+
+    String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_INCORRECT_PARTITIONENDDATE)
+            .replaceAll("TABLENAME", tableFullName);
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("partitionTo").getText();
+    System.out.println(actualErrorMessage);
+    Assert.assertEquals("Error message mismatch for Partition End Date", expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("partitionTo"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Validate BigQuery source incorrect property error for reference name{string} value {string}")
+  public void validateBigQuerySourceIncorrectPropertyErrorForreferncename(String property, String value) {
+    CdfBigQueryPropertiesActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfBigQueryPropertiesLocators.getSchemaButton, 5L);
+    String tableFullName = StringUtils.EMPTY;
+    if (property.equalsIgnoreCase("dataset")) {
+      tableFullName = PluginPropertyUtils.pluginProp("projectId") + ":" + PluginPropertyUtils.pluginProp(value)
+              + "." + TestSetupHooks.bqSourceTable;
+    } else if (property.equalsIgnoreCase("table")) {
+      tableFullName = PluginPropertyUtils.pluginProp("projectId") + ":"
+              + PluginPropertyUtils.pluginProp("dataset")
+              + "." + PluginPropertyUtils.pluginProp(value);
+    } else if (property.equalsIgnoreCase("datasetProject")) {
+      tableFullName = PluginPropertyUtils.pluginProp(value) + ":" + PluginPropertyUtils.pluginProp("dataset")
+              + "." + TestSetupHooks.bqSourceTable;
+    } else if (property.equalsIgnoreCase("referenceName")) {
+      tableFullName = PluginPropertyUtils.pluginProp(value) + ":" + PluginPropertyUtils.pluginProp("reference")
+              + "." + TestSetupHooks.bqSourceTable;
+    }
+    String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_INCORRECT_REFERENCENAME)
+            .replaceAll("TABLENAME", tableFullName);
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("referenceName").getText();
+
+    Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement
+            ("referenceName"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+
+  }
+
+  @Then("Enter BigQuery source properties filter")
```

Comment:
All the step definitions added here in base file are all not needed. Use the existing steps, and execute the tests.